flymake-vale.el

```diff
@@ -139,6 +139,34 @@ Passing the results and source BUF to CALLBACK."
   (or  (string-equal event "finished\n")
        (string-match "exited abnormally with code 1.*" event)))
 
+(defun flymake-vale--guess-extension-from-mode (mode)
+  "This is not good enough. At all.
+
+I don't know of a good way of getting the likely file extension of a
+major mode. The closest I found was the `auto-mode-alist' var, which
+has a regex value for all the file extensions which might match.
+
+Ideally, there'd be a way to generate a valid extension from the
+regexp, or there'd be a way of deciding a default suggested extension
+for a major-mode, but instead of wading in that right now and to prove
+the concept instead, I used this hack."
+  (let ((match (rassoc mode auto-mode-alist)))
+    (cond ((null match) ".txt")
+          (t (string-replace "?" "" (string-replace "'" "" (string-replace "\\" "" (car match))))))))
+
+(defun flymake-vale--detect-extension ()
+  "Attempt to detect a file extension related to the buffer we wish to
+check, either using the file extension or the major mode."
+  (with-current-buffer flymake-vale--source-buffer
+    (let ((file-name (buffer-file-name)))
```

Comment:
read the docstring for `buffer-file-name`. `C-h f buffer-file-name` You can simplify this function.

Reply:
So the reason why I used `with-current-buffer` instead of passing the `flymake-vale--source-buffer` into `buffer-file-name`, is because I wanted to also retrieve `major-mode` from the same buffer - when I run `C-h v major-mode`, it seems like its manipulated by C code in Emacs?

Is there a way around this I'm missing?

Reply:
Yes, you can use the `buffer-local-value` to determine the variable given a specific buffer.